### PR TITLE
rgw: fix break inside of yield in RGWFetchAllMetaCR

### PR DESCRIPTION
**Pull request by cbodley**
the yield macro is implemented with for/switch, so the breaks in
RGWFetchAllMetaCR weren't being applied to the for loop as expected -
so any of these breaks send RGWFetchAllMetaCR into an infinite loop

changed the breaks into returns, so the next call to operate() will
continue with the next iteration of the loop

Fixes: http://tracker.ceph.com/issues/17655

(exposed by unit tests in https://github.com/ceph/ceph/pull/11585)

### CODE REVIEW

**Comment by dachary** [comment]
jenkins test this please (no log)

**Comment by dachary** [comment]
jenkins test this please (jenkins failure)

**Comment by dachary** [comment]
jenkins test this please (eio now ignored in master)

File: src/rgw/rgw_sync.cc [hunk 1]
```diff
@@ -840,7 +840,7 @@ class RGWFetchAllMetaCR : public RGWCoroutine {
           yield {
             if (!lease_cr->is_locked()) {
               lost_lock = true;
-              break;
+              return 0;
```
Comment (cbodley) [inline src/rgw/rgw_sync.cc]: will try replacing the yield {} block with a single `yield;` statement - then the `break`s will apply to the for loop as intended

**Comment by cbodley** [comment]
@yehudasa removed the `yield {}` block and added a single `yield;` statement once per iteration of the loop (i put the yield before the local variables so it wouldn't complain about jumping across their declarations)

i rebased the unit tests from https://github.com/ceph/ceph/issues/11585 for testing - without this pr, they hit an infinite loop - with this pr, they pass

**Comment by yehudasa** [comment]
@cbodley should also test the multisite suite

**Comment by cbodley** [comment]
passes test_multi.py (requires fixes from https://github.com/ceph/ceph/pull/12229 and https://github.com/ceph/ceph/pull/12105)

**Comment by yehudasa** [comment]
@cbodley need to get this one through teuthology too

**Comment by cbodley** [comment]
@yehudasa success http://pulpito.ceph.com/cbodley-2017-01-17_15:16:18-rgw-wip-cbodley-testing---basic-mira

**Pull request merged**